packages/field/src/PrimeField.ts

```diff
@@ -0,0 +1,1444 @@
+/// source: https://github.com/GuildOfWeavers/galois
```

Comment:
We'll probably want to vendor new versions of this library eventually; IMO it'd be very helpful to include some instructions on how to update this library. (i.e. grab the repo, copy these parts over, make these manual changes, ...)

An automated script to "regenerate" the vendored library would be ideal, but depending on how complex the vendoring process is, may be more work than is warranted. And it'd be easy for updates to the galois library to break an automated script.

Reply:
I think we should consider this a hard fork, and plan to independently maintain all of the code in perpetuity. Part of the motivation behind this change was realizing we were going to end up doing that either way, since upstream seems mostly unmaintained. Part of that should probably be removing any code we don't use, and making sure we have better test coverage than in this PR

Reply:
Thinking about this some more, I think you're right that this is the best option. Needing to maintain a finite-field library is not ideal, but given that upstream is unmaintained, we might not have much choice.